src/VersionTool/Program.cs

```diff
@@ -225,13 +253,324 @@ private static int OnList(CommandOption pathOption)
+
+            var repos = (patchConfig["Repos"] as JArray)?.Children()
+                .Select(token => new Repository((string)token, Path.Combine(directory, (string)token)))
+                .ToList();
+
+            // Handle patch config update
+            if (udpatePatchConfigOption.HasValue())
+            {
+                // Clear the Packages section
+                patchConfig["Packages"] = null;
+
+                // Scan for all packages and create new Packages section
+                UpdatePatchConfig(patchConfig, repos);
+                SaveJObject(patchConfigPath, patchConfig);
+
+                return 0;
+            }
+
+            var rules = (patchConfig["Rules"] as JArray)?.Children()
+                .Select(r => (r as JObject).ToObject<Rule>())
+                .ToList();
+            var packages = (patchConfig["Packages"] as JArray)?.Children()
+                .Select(r => (r as JObject).ToObject<Package>())
+                .ToList();
+
+            BuildRepoGraph(repos);
+
+            // Handle package updates
+            if (updatePackageOption.HasValue())
+            {
+                foreach (var packageUpdate in updatePackageOption.Values)
+                {
+                    var parsedPackageUpdate = packageUpdate.Split(new[] { ';' }, 2);
+
+                    if (parsedPackageUpdate.Length != 2)
+                    {
+                        throw new Exception($"Invalid package update option {packageUpdate}.");
+                    }
+
+                    var packageName = parsedPackageUpdate[0];
+                    var newPackageVersion = parsedPackageUpdate[1];
+
+                    var repo = repos.Where(r => r.Packages.Contains(packageName)).SingleOrDefault();
+                    if (repo != null)
+                    {
+                        // update version
+                        OnUpdateVersion(
+                            new List<string>(new[] { Path.Combine(repo.Path, "src", packageName) }),
+                            new List<string>(),
+                            parsedPackageUpdate[1]);
+
+                        // update entry in packages
+                        packages.Where(p => string.Equals(p.Name, packageName, StringComparison.OrdinalIgnoreCase)).Single().NewVersion = newPackageVersion;
+                    }
+                    else
+                    {
+                        // update dependency
+                        OnUpdateDependency(
+                            new List<string>(new[] { directory }),
+                            new List<string>(),
+                            parsedPackageUpdate[0],
+                            parsedPackageUpdate[1]);
+                    }
+                }
+            }
+
+            // Handle repo updates
+            Console.WriteLine("Update repo");
+            if (updateRepoOption.HasValue())
+            {
+                foreach (var repoUpdate in updateRepoOption.Values)
```

Comment:
For multiple repo updates, just run the tool again with the new repo as an argument.